plugins/woocommerce-admin/client/launch-your-store/settings/slotfill.js

```diff
@@ -26,13 +27,40 @@ const SiteVisibility = () => {
 		comingSoon: initialComingSoon = false,
 		storePagesOnly: initialStorePagesOnly = false,
 		privateLink: initialPrivateLink = false,
+		shareKey,
 	} = useLaunchYourStore();
 	const [ comingSoon, setComingSoon ] = useState( initialComingSoon );
 	const [ storePagesOnly, setStorePagesOnly ] = useState(
 		initialStorePagesOnly
 	);
 	const [ privateLink, setPrivateLink ] = useState( initialPrivateLink );
 
+	const copyLink = __( 'Copy link', 'woocommerce' );
+	const copied = __( 'Copied!', 'woocommerce' );
+	const [ copyLinkText, setCopyLinkText ] = useState( copyLink );
+
+	const getPrivateLink = () => {
+		return (
+			window?.wcSettings?.homeUrl +
+			( storePagesOnly === 'yes' ? 'store?woo-share=' : '?woo-share=' ) +
+			shareKey
+		);
+	};
+
+	const copyToClipboard = ( textToCopy ) => {
```

Comment:
I think it'll be good to move this to a utils/helper file

Reply:
There is a hook for this in the `@wordpress/compose` package. We can probably use that instead of writing our own.

https://github.com/woocommerce/woocommerce/blob/0cb3b4a9ad09acf97a56ac474e1431ece9ca30e7/packages/js/product-editor/src/components/iframe-editor/header-toolbar/more-menu/copy-all-content-menu-item.tsx#L7

Reply:
Clicking behavior is fixed in https://github.com/woocommerce/woocommerce/pull/45915/commits/7b620af67f520b3ba6d890fba030f6a157227120

> (Optional) It seems like when woocommerce_share_key option doesn't exist, the share key is false. Should we address generating a random share key in this PR? I don't have a strong preference.

I think once we enable feature flag, new installation will have the option from https://github.com/woocommerce/woocommerce/pull/45306

Reply:
> When clicking on the new component, the checkbox toggle is triggered. I think we should change it to copy instead. What do you think?

@ilyasfoo I've tested it after the update and I actually think this is intended feature and I find it convenient and we should keep it as it is to keep the consistency. What do you think?

Reply:
@moon0326 Personally I feel it's a different behaviour than UIs that I'm used to. Just making sure we're on the same page, I'm referring to clicking the link in the white component, where I was expecting it to copy the link instead of disabling toggle:
 
<img width="701" alt="image" src="https://github.com/woocommerce/woocommerce/assets/3747241/abe264c4-5310-447c-b8cb-64f2c68e8f39">

Reply:
@ilyasfoo Ah..I thought we were talking about the text area. Yup! I agree that the white box should not trigger the toggle. I'll handle it in a separate PR.